### ISSUE TITLE
"Global.Unexpected_None" exception thrown on first invocation of GenProg; no exception thrown on subsequent invocations

### ISSUE THREAD

**Issue by ChrisTimperley** [squaresLab/genprog-code#9]
```
$ pip3 install --upgrade bugzoo
$ bugzoo source add https://github.com/squaresLab/genprog-code
$ bugzoo source add https://github.com/squaresLab/ManyBugs
$ bugzoo source update
$ bugzoo tool build -f genprog
$ bugzoo bug build -f manybugs:python:69224-69224
$ bugzoo container launch manybugs:python:69224-69224 --with genprog
> genprog configuration-default --skip-failed-sanity-tests --allow-coverage-fail
...
...
...
test: 284
cilRep: default.cache: saved
cachingRep: default.cache: saved
faultlocRep: default.cache: saved
cilrep done serialize
cilRep: stmt_count = 1062
cilRep: stmts in weighted_path = 215
cilRep: total weight = 9.4
cilRep: stmts in weighted_path with weight >= 1.0 = 1
cilRep: Modules/selectmodule.c (code bank/base file; atoms [1,1156])
cilRep: 1 file(s) total in representation
search: genetic algorithm begins (|original| = 0.493019 MB)
cilRep: shared size: 0.558815 MB
cilRep: private size: 0.0123653 MB
search: initial population (sizeof one variant = 0.49304 MB)
genprog aborting: Global.Unexpected_None
Rep: saving test cache
Variant Test Case Queries: 0
"Test Suite Evaluations": 0

Compile Failures: 0
Wall-Clock Seconds Elapsed: 1311.11
Activity Name                     Count Seconds = Percent of Total Time
  compile                               2 104.407 = 7.97419%
  test                                570 1204.842 = 92.0208%
  TOTAL                                   1309.315 = 99.8625% (avg CPU usage)
Fatal error: exception Global.Unexpected_None
```

If the user executes the same command once again, then no exception is thrown and the repair process proceeds as expected.

**Comment by joerenzullo** [comment]
I got the same error with an different Python bug this afternoon - 69934-69935

**Comment by yodahaji** [comment]
I got same error with fedora 29.

**Comment by pdreiter** [comment]
Identified the solution to this bug:
Line 1159: [Minor Consistency issue] Ensured that Negative tests precede Positive test evaluations
Line 1363: [Root-Cause of Global.Unexpected_None Exception] Ensures that '**cached information**' infrastructure is reset before test_cases are evaluated.
CAVEAT: There could be unintended side-effects to the complete reset of the **cached information** - BUT this is consistent with the **brute force** search, where `eval_count` is reset due to the calls to mutation operators.
Minimal fix is just : `eval_count <- TestMap.empty, 0 ;` rather than `self#update () ;`

[rep.ml.diff.txt](https://github.com/squaresLab/genprog-code/files/2833816/rep.ml.diff.txt)

**Comment by jlacomis** [comment]
Fixed by #14
